src/chunk_scan.c

```diff
@@ -179,10 +186,17 @@ ts_chunk_scan_by_constraints(const Hyperspace *hs, const List *dimension_vecs,
 		MemoryContextSwitchTo(orig_mcxt);
 		MemoryContextDelete(work_mcxt);
 
+		Chunk *osm_chunk = NULL;
+		if (osm_chunk_id != INVALID_CHUNK_ID)
+		{
+			osm_chunk = ts_chunk_get_by_id(osm_chunk_id, true);
+			chunks = MemoryContextAlloc(orig_mcxt, sizeof(Chunk *) * 1);
+			chunks[0] = osm_chunk;
+		}
 		if (numchunks)
-			*numchunks = 0;
+			*numchunks = osm_chunk ? 1 : 0;
```

Comment:
Perhaps for code readability you can introduce a constant `#define MAX_OSM_CHUNKS  1` and use it here and also in the multiplication you made before for memory allocation... I know it's very simple but as far I understood this max number of OSM chunks is part of the design, right?

Reply:
yes. max # of chunks is 1. This code block is going away soon when the new chunk exclusion logic gets merged.